homeassistant/components/recorder/auto_repairs/statistics/schema.py

```diff
@@ -1,89 +1,43 @@
+from ...db_schema import (
+    DOUBLE_PRECISION_TYPE_SQL,
+    Statistics,
+    StatisticsMeta,
+    StatisticsShortTerm,
+)
 from ...models import StatisticData, StatisticMetaData, datetime_to_timestamp_or_none
 from ...statistics import (
     _import_statistics_with_session,
     _statistics_during_period_with_session,
 )
 from ...util import session_scope
+from ..schema import (
+    PRECISE_NUMBER,
+    check_columns,
+    correct_table_character_set_and_collation,
+    get_precise_datetime,
+    validate_table_schema_supports_utf8,
+)
 
 if TYPE_CHECKING:
     from ... import Recorder
 
 _LOGGER = logging.getLogger(__name__)
 
+STATISTIC_ID = f"{DOMAIN}.db_test"
 
-def _validate_db_schema_utf8(
-    instance: Recorder, session_maker: Callable[[], Session]
-) -> set[str]:
-    """Do some basic checks for common schema errors caused by manual migration."""
-    schema_errors: set[str] = set()
-
-    # Lack of full utf8 support is only an issue for MySQL / MariaDB
-    if instance.dialect_name != SupportedDialect.MYSQL:
-        return schema_errors
-
-    # This name can't be represented unless 4-byte UTF-8 unicode is supported
-    utf8_name = "𓆚𓃗"
-    statistic_id = f"{DOMAIN}.db_test"
-
-    metadata: StatisticMetaData = {
-        "has_mean": True,
-        "has_sum": True,
-        "name": utf8_name,
-        "source": DOMAIN,
-        "statistic_id": statistic_id,
-        "unit_of_measurement": None,
-    }
-    statistics_meta_manager = instance.statistics_meta_manager
-
-    # Try inserting some metadata which needs utf8mb4 support
-    try:
-        # Mark the session as read_only to ensure that the test data is not committed
-        # to the database and we always rollback when the scope is exited
-        with session_scope(session=session_maker(), read_only=True) as session:
-            old_metadata_dict = statistics_meta_manager.get_many(
-                session, statistic_ids={statistic_id}
-            )
-            try:
-                statistics_meta_manager.update_or_add(
-                    session, metadata, old_metadata_dict
-                )
-                statistics_meta_manager.delete(session, statistic_ids=[statistic_id])
-            except OperationalError as err:
-                if err.orig and err.orig.args[0] == 1366:
-                    _LOGGER.debug(
-                        "Database table statistics_meta does not support 4-byte UTF-8"
-                    )
-                    schema_errors.add("statistics_meta.4-byte UTF-8")
-                    session.rollback()
-                else:
-                    raise
-    except Exception as exc:  # pylint: disable=broad-except
-        _LOGGER.exception("Error when validating DB schema: %s", exc)
-    return schema_errors
-
-
-def _get_future_year() -> int:
-    """Get a year in the future."""
-    return datetime.now().year + 1
+DOUBLE_PRECISION_COLUMNS = ("mean", "min", "max", "state", "sum")
 
 
 def _validate_db_schema(
```

Comment:
This whole function can be replaced with the one for states with changing the table and column names